The backend does not require a mesonbuild module for every flavor.
`meson` and `ninja` commands from the regular RPM packages are
sufficient.

Index: meson_python-0.8.1/pyproject.toml
===================================================================
--- meson_python-0.8.1.orig/pyproject.toml
+++ meson_python-0.8.1/pyproject.toml
@@ -26,8 +26,6 @@ classifiers = [

 dependencies = [
   'colorama; os_name == "nt"',
-  'meson>=0.62.0',
-  'ninja',
   'pyproject-metadata>=0.5.0', # not a hard dependency, only needed for projects that use PEP 621 metadata
   'tomli>=1.0.0',
   'typing-extensions>=3.7.4; python_version<"3.8"',
